lib/features/auth/data/repositories/user_repository.dart

```diff
@@ -81,4 +81,27 @@ class UserRepository {
       return left(Failure.internalServerError(message: e.toString()));
     }
   }
+
+  Either<Failure, WeightUnit> updateWeightUnit({
+    required WeightUnit unit,
+  }) {
+    try {
+      final res = box.getAll();
+
+      if (res.isEmpty) {
+        return left(const Failure.empty());
+      }
+
+      final user = res.first;
+
+      box.put(
+        user..dbPreferredWeightUnit = unit.index,
```

Comment:
You don't have to use the dbPreferredWeightUnit here, you can use the transient preferredWeightUnit instead.

This saves you from doing the index call.

Reply:
updated user preferred theme and weight units to use the enums